sw/device/tests/alert_handler_lpg_clkoff_test.c

```diff
@@ -612,5 +569,91 @@ bool test_main(void) {
+
+    The SystemVerilog test sequence associated with this test is located here:
+      - hw/top_earlgrey/dv/env/seq_lib/chip_sw_alert_handler_lpg_clkoff_vseq.sv
+
+    TEST:
+    for_each_peripheral
+      1- Wait for random time
+      2- Enable the peripheral's clock
+      3- Trigger the peripheral's fatal alert
+      4- Disable the peripheral's clock
+      5- Wait for 1ms
+      6- Enable the peripheral's clock
+      7- Confirm that fatal_alert is still there
+  */
+
+  // TODO: Test all peripherals in parallel instead of serial to further improve
+  // the runtime.
+  while (test_step_cnt < 3 * ARRAYSIZE(kPeripherals)) {
+    // Run this test phase only in DV sim
+    if (kDeviceType == kDeviceSimDV) {
+      // Read the test_step_cnt and compute the test phase
+      // amd the peripheral ID to test
+      test_phase = test_step_cnt / ARRAYSIZE(kPeripherals);
+      peri_idx = test_step_cnt - (test_phase)*ARRAYSIZE(kPeripherals);
+
+      // Wait for a random time <= 1024 cycles
+      busy_spin_micros(rand_testutils_gen32_range(1, 1024));
+
+      // Enable the clock of the peripheral just in case
+      set_peripheral_clock(&kPeripherals[peri_idx], kDifToggleEnabled);
+      // Wait for 100us
+      busy_spin_micros(100);
+
+      // Anchor log monitored by the SV test sequence. Do not modify.
+      LOG_INFO("Peripheral %0d ready for fault injection", peri_idx);
+
+      // The SV test sequence will at this point, inject a fatal alert into
+      // this peripheral. Wait for the alert to trigger.
+      IBEX_SPIN_FOR(alert_handler_testutils_is_alert_active(
+                        &alert_handler, kPeripherals[peri_idx].alert_ids[0]),
+                    100 /*us*/);
+
+      // Disable the clock of the peripheral
+      set_peripheral_clock(&kPeripherals[peri_idx], kDifToggleDisabled);
+
+      // Clear the alert_handler.ALERT_CAUSE bit before re-enabling the clocks
+      CHECK_DIF_OK(dif_alert_handler_alert_acknowledge(
+          &alert_handler, kPeripherals[peri_idx].alert_ids[0]));
+      // Verify that ALERT_CAUSE is actually cleared
+      CHECK_DIF_OK(dif_alert_handler_alert_is_cause(
+          &alert_handler, kPeripherals[peri_idx].alert_ids[0], &is_cause));
+      CHECK(!is_cause, "is_cause for alert[%d] should be 0 but we got 1",
+            kPeripherals[peri_idx].alert_ids[0]);
+
+      // Wait long enough time to leave the clock gating enabled. Arbitrarily
+      // set to 500us to keep the runtime reasonable.
+      busy_spin_micros(500);
+
+      // Enable the clock of the peripheral
+      set_peripheral_clock(&kPeripherals[peri_idx], kDifToggleEnabled);
+      busy_spin_micros(100);
+
+      // Check if the fatal alert is still there
+      CHECK_DIF_OK(dif_alert_handler_alert_is_cause(
+          &alert_handler, kPeripherals[peri_idx].alert_ids[0], &is_cause));
+      CHECK(is_cause, "is_cause should be 1 but we got %d", is_cause);
+      // Increment the test counter
+      test_step_cnt++;
+    } else {
```

Comment:
We can remove the `else` path after making sure the test passes in DV. I put this one just to see if the test_step_cnt is incremented correctly on FPGA.

Reply:
Done.